contracts/router/Adapter.sol

```diff
@@ -57,26 +57,27 @@ abstract contract Adapter is Ownable, IAdapter {
 
     /**
      * @notice Recover ERC20 from contract
-     * @param _tokenAddress token address
-     * @param _tokenAmount amount to recover
+     * @param _token token to recover
      */
-    function recoverERC20(address _tokenAddress, uint256 _tokenAmount)
-        external
-        onlyOwner
-    {
-        require(_tokenAmount > 0, "Adapter: Nothing to recover");
-        emit Recovered(_tokenAddress, _tokenAmount);
-        IERC20(_tokenAddress).safeTransfer(msg.sender, _tokenAmount);
+    function recoverERC20(IERC20 _token) external onlyOwner {
```

Comment:
@ChiTimesChi having a "recover all" function is a good move, but I am curious why you removed the ability to recover specific amounts entirely -- if there's no reason to have that then disregard me and carry on

Reply:
@bleaker0x Adapter isn't suited for storing tokens. Some of the adapters (Curve, Synapse, ...) are designed in a way, where you have to transfer token directly to Adapter contract, and then invoke `swap()` (all in the same tx to avoid griefing). This means that if Adapter has a supported token lying around, anyone can invoke `swap()` without transferring first, effectively "rescuing" the token.